### PR TITLE
Profile/aw/65410/disallow letters in phone fields

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- Changes the validation rules of the phone field to only allow numbers, -, and spaces in the field. This was the field will show a validation error if any invalid characters are input

- Error message content was updated per the ticket as well 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/65410

## Testing done

- Existing tests function appropriately

## Screenshots

No UI changes

## What areas of the site does it impact?

Profile

## Acceptance criteria

- [x] Validation regex updated for allowing only numbers, dashes, and spaces

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: src/applications/personalization/profile/util/contact-information/phoneUtils.js [hunk 1]
```diff
@@ -11,7 +10,8 @@ export const phoneFormSchema = {
     },
     inputPhoneNumber: {
       type: 'string',
-      pattern: '^\\d{10}$',
+      pattern: '^[0-9- ]+$',
```
Comment (allisonlu) [inline src/applications/personalization/profile/util/contact-information/phoneUtils.js:13]: Should we check for parentheses too?

**Review by allisonlu — APPROVED** [review]
:shipit:

File: src/applications/personalization/profile/util/contact-information/phoneUtils.js (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/personalization/profile/util/contact-information/phoneUtils.js]: I added a check for parens @allisonlu thanks for pointing that out. I initially tried to pattern match to make sure the parens had 3 digits between but couldn't get that to work, so I just added parens as accepted chars in the regex.

good question, the code that removes the special characters is used when submitting the field to the api. It splits up the number and area code as well before sending to the api. @dellerbie

Comment (adamwhitlock1) [reply src/applications/personalization/profile/util/contact-information/phoneUtils.js]: If we allowed anything, then yes the stripping would get rid of any non-digits before sending to the api, but it was causing an issue with the form field being dirty / not dirty in regards to the "Edit" page usage of the phone field.

There are lost of weird issues like this that I am finding with using the form library.

**Pull request merged**